playwright/_impl/_browser_type.py

```diff
@@ -135,6 +136,39 @@ async def launch_persistent_context(
                 raise not_installed_error(f'"{self.name}" browser was not found.')
             raise e
 
+    async def connect(
+        self, ws_endpoint: str, timeout: float = None, slow_mo: float = None
+    ) -> Browser:
+        transport = WebSocketTransport(ws_endpoint, timeout)
+
+        connection = Connection(
+            self._connection._dispatcher_fiber,
+            self._connection._object_factory,
+            transport,
+        )
+        connection._is_sync = self._connection._is_sync
+        connection._loop = self._connection._loop
+        connection._loop.create_task(connection.run())
+        self._connection._child_ws_connections.append(connection)
+        playwright = await connection.wait_for_object_with_known_name("Playwright")
+        pre_launched_browser = playwright._initializer.get("preLaunchedBrowser")
+        assert pre_launched_browser
+        browser = cast(Browser, from_channel(pre_launched_browser))
+        browser._is_remote = True
```

Comment:
Now that you are adding it, we need to introduce the code that depends on it. That would be our artifacts, namely 'download' and 'video'.

Reply:
good catch, added it.